client/components/chart/index.jsx

```diff
@@ -107,8 +108,12 @@ function Chart( {
 
 	const width = isTouch && sizing.clientWidth <= 0 ? 350 : sizing.clientWidth - chartXPadding; // mobile safari bug with zero width
 	const maxBars = Math.floor( width / minWidth );
+
 	// Memoize data calculations to avoid performing them too often.
 	const { chartData, isEmptyChart, yMax } = useMemo( () => {
+		// We communicate the new value for maximum bars in the chart to the parent component
+		onChangeMaxBars( maxBars );
```

Comment:
Very nice solution! I

Reply:
You might want to check if it's defined, though. And also pass it to the deps array (to keep ESLint from complaining).